File: Easy/registryclass/meta_registryclasspersoneview/meta_registryclasspersoneview.cs
/*
    Easy
    Copyright (C) 2020 Università degli Studi di Catania (www.unict.it)
    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.
    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.
    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/

﻿using System;
using System.Data;
using metadatalibrary;
using metaeasylibrary;
//$CustomUsing$


namespace meta_registryclasspersoneview
{
    public class Meta_registryclasspersoneview : Meta_easydata 
	{
        public Meta_registryclasspersoneview(DataAccess Conn, MetaDataDispatcher Dispatcher) :
            base(Conn, Dispatcher, "registryclasspersoneview") {
				Name = "Tipologia fiscale";
			EditTypes.Add("persone");
			ListingTypes.Add("persone");
			//$EditTypes$
        }

		private string[] mykey = new string[] {"idregistryclass"};

		public override string[] primaryKey() {
			return mykey;
		}

		//$SetDefault$

		//$Get_New_Row$

		public override bool IsValid(DataRow R, out string errmess, out string errfield) {
			if (!base.IsValid(R, out errmess, out errfield)) return false;

			//$IsValid$

			return true;
		}

		public override void DescribeColumns(DataTable T, string ListingType) {
			base.DescribeColumns(T, ListingType);

			foreach (DataColumn C in T.Columns) {
				DescribeAColumn(T, C.ColumnName, "", -1);
			}
			int nPos = 1;
					
			switch (ListingType) {
				case "persone": {
						DescribeAColumn(T, "idregistryclass", "Codice", nPos++);
						DescribeAColumn(T, "description", "Descrizione", nPos++);
						DescribeAColumn(T, "registryclass_active", "Attivo", nPos++);
						break;
					}
					//$DescribeAColumn$
			}
		}

		//$GetSortings$

		public override string GetStaticFilter(string ListingType) {
			switch (ListingType) {
				//$GetStaticFilter$
			}
			return base.GetStaticFilter(ListingType);
		}

		//$CustomCode$
	}
}

File: Easy/Portale/client/app_segreterie/metadata/registryclass/meta_registryclasspersoneview.js
(function() {

    var MetaData = window.appMeta.MetaSegreterieData;

    function meta_registryclasspersoneview() {
        MetaData.apply(this, ["registryclasspersoneview"]);
        this.name = 'meta_registryclasspersoneview';
    }

    meta_registryclasspersoneview.prototype = _.extend(
        new MetaData(),
        {
            constructor: meta_registryclasspersoneview,
			superClass: MetaData.prototype,

			describeColumns: function (table, listType) {
				var nPos=1;
				var self = this;
				_.forEach(table.columns, function (c) {
					self.describeAColumn(table, c.name, '', null, -1, null);
				});
				switch (listType) {
					default:
						return this.superClass.describeColumns(table, listType);
					case 'persone':
						this.describeAColumn(table, 'idregistryclass', 'Codice', null, 10, 2);
						this.describeAColumn(table, 'description', 'Descrizione', null, 20, 150);
						this.describeAColumn(table, 'registryclass_active', 'Attivo', null, 30, null);
//$objCalcFieldConfig_persone$
						break;
//$objCalcFieldConfig$
				}
				return appMeta.Deferred("describeColumns").resolve();
			},


			//$setCaptions$

			primaryKey: function () {
				return ["idregistryclass"];
			},


			//$isValidFunction$

			//$getStaticFilter$

			//$getSorting$

        });

    window.appMeta.addMeta('registryclasspersoneview', new meta_registryclasspersoneview('registryclasspersoneview'));

	}());
